DB.java

```diff
@@ -294,6 +294,85 @@ else if(i==0)
 		
 		return result;
 	}
+	
+	public boolean check_comments_popular(String ip, int id, int a) throws SQLException
+	{
+		connect();
+		
+		boolean result = false;
+		int i = 0;
 
-}
+		
+
+		String sql = "insert into comments_ip(ip, id, prefer) values(?, ?, ?)";
+		String sql2 = "select * from comments_ip where ip=? and id=?";
+		String sql3 = "update comments_ip set prefer=? where ip=? and id=?";
+		
+		PreparedStatement ps2 = con.prepareStatement(sql2);
+		ps2.setString(1, ip);
+		ps2.setInt(2, id);
+		
+		ResultSet rs = ps2.executeQuery();
+		
+		while(rs.next())

```

Comment:
@mjhmohamed 

What is the purpose of this while loop?  Add lots of comments.

**Hui**

Reply:
Actually this while loop is quite standard when handling Java SQL ResultSet.

Simply, `rs.next()` will return true, iff the query executed successfully and there is results forward(top-bottom approach) in the set of records retrieved by the sql query.
Otherwise, it will return false.

IBRAHIM.